### DISCUSSION TITLE
Code Challenge Submission for "[4] Collection and Study Management"

### DISCUSSION THREAD

**Discussion by CoToYo** [camicroscope/GSOC#26] [Q&A]
Dear mentors,

I am a master student majoring in Computer Science in Boston University.

After my seriously consideration and study, I was really excited and interested about one of our great projects "[4] Collection and Study Management". 
I noticed that there was a code challenge about NodeJS, and I was happened to be working on a related project by using NodeJS, Express and MongoDB.

So I finished it as a simple version and upload the code to my github repository.

demo: https://nice-blog.onrender.com/

code link: https://github.com/CoToYo/NiceBlog

video link: https://www.youtube.com/watch?v=9MQK54GPSJE

I used NodeJS, Express and MongoDB and built a simple blog posting web application. To improve the implementation, I designed the code into "MVC" structure and built/used various middleware to power the development process.

Thank you so much for taking your time to have a look.
I will start to work on my proposal, and I really expected to hear any feedback from you!

Thanks again.

Tingyuan Cui

**Comment by CoToYo** [comment]
I just added a demo video for this app, if anyone of you have interested in it, please check it out!💪

**Reply by CoToYo** [reply]
Just deploy a demo application! Please check it out!

**Reply by nanli-emory** [reply]
Hi @CoToYo sorry for the delay in responding. I went through your demo. It is a good one but still needs to polish. Do you submit your proposal?

**Reply by CoToYo** [reply]
@nanli-emory Thank you so much for your response!!!🥹

I haven't submitted the proposal yet but I will try to complete and submit it as soon as possible. Once I complete it, I would send you for review and feedback.

And may I ask for any suggestion on improving my code challenge demo? Although there is not much time left now, I will give a try to improve it!

**Reply by nanli-emory** [reply]
For example, input data validation, Error messages, and UX things at the frontend.
I think you can keep it as a draft and I can review it but don't forget to submit it as a proposal before the deadline.

**Reply by CoToYo** [reply]
OK! So for the proposal, I will put it on Google document for your review after I finish the first draft. 
And for the improvement of my code challenge demo, I will work on it after the completion of the first draft.

I apologize for the delay since I am also preparing for the midterm this week. 
Please don't worry, I will finish it as soon as possible. Thank you for you understanding!!😉

**Reply by shaiktahseen** [reply]
Hi camicroscope community, I'm Shaik Tahseen. I'm a beginner and have knowledge in Html, CSS, JavaScript & Flexboxes, and I'm curious to join GSOC 2023 this year.
I like to contribute Can anyone help me to know how to join Jenkins & how to contribute & get into camicroscope  as a contributor with basic skills?  I really like to learn, contribute, give my best & look for opportunities where I can upskill myself & also fulfill the goals of the organization.

If anyone let me know what would be really helpful.
I see the closing date for GSOC is 4th April 2023

Here are my bootcamp certifications.
HTML & CSS Essentials Bootcamp
https://verify.letsupgrade.in/certificate/LUEHTMLFEB123307

JavaScript Programming Essentials Bootcamp
https://verify.letsupgrade.in/certificate/LUEJSMAR123657

•	Email : shaiktahseen05@gmail.com
•	GitHub :  https://github.com/shaiktahseen
•	LinkedIn :  https://www.linkedin.com/in/shaik-tahseen-%F0%9F%87%AE%F0%9F%87%B3-39074b17a/
•	Twitter: https://twitter.com/shaiktahseen\_05

**Reply by CoToYo** [reply]
> Hi @CoToYo sorry for the delay in responding. I went through your demo. It is a good one but still needs to polish. Do you submit your proposal?

@nanli-emory I have submitted the draft of my proposal for "Collection and Study Management" project.

I know there are a lot of excellent proposals from other contributors waited for you to review, but please take a look at my proposal and give some feedback if available. 

Thank you so much for taking your time to look at it.

**Reply by CoToYo** [reply]
@nanli-emory Just polished my code challenge by  adding complete user input validation and error message. Please check it out.😄

some examples:
<img width="946" alt="image" src="https://user-images.githubusercontent.com/56789038/229336052-4d241471-6852-4516-8941-5f5e23bddf91.png">
<img width="960" alt="image" src="https://user-images.githubusercontent.com/56789038/229336245-8dfe4db0-6ef5-4bb8-ba58-4dd925ffca15.png">

**Reply by shaiktahseen** [reply]
> @nanli-emory Just polished my code challenge by  adding complete user input validation and error message. Please check it out.😄
> 
> some examples:
> <img width="946" alt="image" src="https://user-images.githubusercontent.com/56789038/229336052-4d241471-6852-4516-8941-5f5e23bddf91.png">
> <img width="960" alt="image" src="https://user-images.githubusercontent.com/56789038/229336245-8dfe4db0-6ef5-4bb8-ba58-4dd925ffca15.png">
> 

@CoToYo hey 👋

I'm Shaik Tahseen. I'm a beginner and have knowledge in Html, CSS, JavaScript and I'm curious to join GSOC 2023 this year.

I like to contribute Can anyone help me to know how to join Jenkins & how to contribute & get into camicroscope as a contributor with basic skills? 

I really like to learn, contribute, give my best & look for opportunities where I can upskill myself & also fulfill the goals of the organization.

If anyone let me know what would be really helpful.

I see the closing date for GSOC is 4th April 2023

• Email : shaiktahseen05@gmail.com
• GitHub : https://github.com/shaiktahseen
• LinkedIn : https://www.linkedin.com/in/shaik-tahseen-%F0%9F%87%AE%F0%9F%87%B3-39074b17a/
• Twitter: https://twitter.com/shaiktahseen\_05

**Reply by CoToYo** [reply]
I believe all the information you need is available at GSOC official website and the github repo maintained by caMicroscope.
some links you may find helpful: https://summerofcode.withgoogle.com/help

And please raise your own discussion channel for your questions or issues, otherwise some important issues and conversations in this discussion channel would be covered up. Thank you.

**Comment by CoToYo** [comment]
Have setup running environment for caMicroscope application and successfully start it.
<img width="1674" alt="image" src="https://user-images.githubusercontent.com/56789038/226805331-bf187b9f-db0c-48aa-ad89-4595284bf98f.png">

**Reply by shaiktahseen** [reply]
Hi camicroscope community, I'm Shaik Tahseen. I'm a beginner and have knowledge in Html, CSS, JavaScript & Flexboxes, and I'm curious to join GSOC 2023 this year.
I like to contribute Can anyone help me to know how to join Jenkins & how to contribute & get into camicroscope  as a contributor with basic skills?  I really like to learn, contribute, give my best & look for opportunities where I can upskill myself & also fulfill the goals of the organization.

If anyone let me know what would be really helpful.
I see the closing date for GSOC is 4th April 2023

Here are my bootcamp certifications.
HTML & CSS Essentials Bootcamp
https://verify.letsupgrade.in/certificate/LUEHTMLFEB123307

JavaScript Programming Essentials Bootcamp
https://verify.letsupgrade.in/certificate/LUEJSMAR123657

•	Email : shaiktahseen05@gmail.com
•	GitHub :  https://github.com/shaiktahseen
•	LinkedIn :  https://www.linkedin.com/in/shaik-tahseen-%F0%9F%87%AE%F0%9F%87%B3-39074b17a/
•	Twitter: https://twitter.com/shaiktahseen\_05
